catalog/cataloger_diff.go

```diff
@@ -245,12 +246,15 @@ func (c *cataloger) diffFromChild(ctx context.Context, tx db.Tx, childID, parent
 	return nil
 }
 
-// contextWithDiffResultsDispose generate diff results id used for temporary table name
-func contextWithDiffResultsDispose(ctx context.Context, tx sq.Execer) (context.Context, context.CancelFunc) {
-	id := xid.New().String()
+// withDiffResultsContext generate diff results id used for temporary table name
+func (c *cataloger) withDiffResultsContext(ctx context.Context) (context.Context, context.CancelFunc) {
+	id := strings.ReplaceAll(uuid.New().String(), "-", "")
```

Comment:
Actually, this produces a valid table name only 37.5% of the time, when the first hex digit is a-f.  You might prepend "diff_table_" to this string.
Or just use nanoid, which is designed to produce useful unescaped names :-/

Reply:
yeah, it already _gets_ a prefix.